qiskit/chemistry/ground_state_calculation/mes_ground_state_calculation.py

```diff
@@ -0,0 +1,178 @@
+    def solver(self) -> Union[MinimumEigensolver, MESFactory]:
+        """Returns the minimum eigensolver or factory."""
+        return self._solver
+
+    @solver.setter
+    def solver(self, solver: Union[MinimumEigensolver, MESFactory]) -> None:
+        """Sets the minimum eigensolver or factory."""
+        self._solver = solver
+
+    def returns_groundstate(self) -> bool:
+        """TODO
+        whether the eigensolver returns the ground state or only ground state energy."""
+
+        return False
+
+    def compute_groundstate(self, driver: BaseDriver,
+                            aux_operators: Optional[List[Any]] = None
+                            ) -> EigenstateResult:
+        """Compute Ground State properties.
+
+        Args:
+            driver: a chemistry driver object which defines the chemical problem that is to be
+                    solved by this calculation.
+            aux_operators: Additional auxiliary operators to evaluate at the ground state.
+                Depending on whether a fermionic or bosonic system is solved, the type of the
+                operators must be ``FermionicOperator`` or ``BosonicOperator``, respectively.
+
+        Raises:
+            NotImplementedError: If an operator in ``aux_operators`` is not of type
+                ``FermionicOperator``.
+
+        Returns:
+            An eigenstate result. Depending on the transformation this can be an electronic
+            structure or bosonic result.
+        """
+        if aux_operators is not None:
+            if any(not isinstance(op, (WeightedPauliOperator, FermionicOperator))
+                   for op in aux_operators):
+                raise NotImplementedError('Currently only fermionic problems are supported.')
+
+        # get the operator and auxiliary operators, and transform the provided auxiliary operators
+        # note that ``aux_operators`` contains not only the transformed ``aux_operators`` passed
+        # by the user but also additional ones from the transformation
+        operator, aux_operators = self.transformation.transform(driver, aux_operators)
+
+        if isinstance(self._solver, MESFactory):
+            # this must be called after transformation.transform
+            solver = self._solver.get_solver(self.transformation)
+        else:
+            solver = self._solver
+
+        # if the eigensolver does not support auxiliary operators, reset them
+        if not solver.supports_aux_operators():
+            aux_operators = None
+
+        raw_mes_result = solver.compute_minimum_eigenvalue(operator, aux_operators)
+
+        eigenstate_result = EigenstateResult()
+        eigenstate_result.raw_result = raw_mes_result
+        eigenstate_result.eigenvalue = raw_mes_result.eigenvalue
+        eigenstate_result.aux_operator_eigenvalues = raw_mes_result.aux_operator_eigenvalues
+        result = self.transformation.interpret(eigenstate_result)
+        return result
+
+    def evaluate_operators(self,
```

Comment:
I commented in the interface that this implements that this seems to be more a standalone utility function. Given aux_operators get evaluated anyway when passed along with the main operator it seems less than cleat why this is in the interface - I get perhaps wanting some standalone function.

Reply:
For the calculation of EOM we need to evaluate aux_operators that come from the ESC interface. We need to be able to evaluate aux_operators even when they are not passed directly with the main_operator

Reply:
Maybe they need evaluating differently than the aux_ops which  are passed to say VQE where it evaluates them which is why its needed. The above comment was more around placement. Its just not clear why some internal need to evaluate operators is part of the interface. This I guess is because of the mix of interface and base class. Is this why AdaptVQE is extending MinEigensolverGroundStateCalculation in order to provide access to this method that is implemented there rather than just implementing the GroundStateCalculation interface that I would expect?

Reply:
I see your point Steve. Do you have a suggestion where to place this method instead?
Its definition will become smth like `evaluate_operators(state, operators, quantum_instance)` since the last arg is the only piece of information which we currently extract out of `self`.

Reply:
Maybe this depends on whatever naming/re-org - if Bosonic/Fermionic operators and transforms are together somewhere maybe together with them? At this point you probably have a better insight than I.

Reply:
Well I couldn't really think of something more sensible than here.
The Bosonic/Fermionic operators are in the root of qiskit.chemistry whereas the transformations have their own module.

I guess we can move this function to a more sensible place once we have converged on the naming and location of all classes/modules. Let's leave this thread open until then :+1:

Reply:
Intuitively it makes most sense to me to have it on the GSC/ESC because they know the runtime environments, as Stefan mentioned [above](https://github.com/Qiskit/qiskit-aqua/pull/1288/files#r503832782). I don't see this as utility to evaluate operators in general, this is straightforward to do, but as a method to allow user to evaluate operators in the same setting as the algorithm.